code/xcvm/lib/core/src/gateway/config.rs

```diff
@@ -260,23 +281,26 @@ pub struct BridgeAsset {
 	pub location_on_network: ForeignAssetId,
 }
 
-/// Definition of an asset on this local chain to operate with
+/// Definition of an asset native to some chain to operate on.
+/// For example for Cosmos CW and EVM chains both CW20 and ERC20 can be actual.
+/// So if asset is local or only remote to some chain depends on context of network or connection.
+/// this design leads to some dummy matches, but in general unifies code (so that if one have to
+/// solve other chain route it can)
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
 #[serde(rename_all = "snake_case")]
 #[cfg_attr(feature = "std", derive(schemars::JsonSchema))]
 pub enum AssetReference {
 	Native { denom: String },
 	Cw20 { contract: Addr },
-	// ADR001 { sha256: cosmwasm_std::HexBinary },
+	Erc20 { contract: EthAddress },
 }
 
 impl AssetReference {
 	pub fn denom(&self) -> String {
 		match self {
-			// AssetReference::ADR001 { sha256 } => ["ibc/",
-			// &sha256.to_string()].concat().to_string(),
 			AssetReference::Native { denom } => denom.clone(),
 			AssetReference::Cw20 { contract } => ["cw20:", contract.as_str()].concat(),
+			AssetReference::Erc20 { contract } => ["erc20:", &contract.to_string()].concat(),
```

Comment:
```suggestion
			AssetReference::Erc20 { contract } => format!("erc20:{contract}"),
```

Reply:
i believe more likely owners will intorduce as_str will introduce sooner or later, when more ibc. so will retain with as easier to refactor to as_str